java/integTest/src/test/java/glide/TransactionTestUtilities.java

```diff
@@ -791,18 +796,94 @@ private static Object[] geospatialCommands(BaseTransaction<?> transaction) {
+                            new GeoSearchShape(200, GeoUnit.KILOMETERS),
+                            new GeoSearchOptions.builder().withhash().withdist().withcoord().build(),
+                            new GeoSearchResultOptions(SortOrder.ASC, 2))
+                    .geosearch(
+                            geoKey1,
+                            new GeoSearchOrigin(new GeospatialData(15, 37)),
+                            new GeoSearchShape(400, 400, GeoUnit.KILOMETERS),
+                            new GeoSearchOptions.builder().withhash().withdist().withcoord().build(),
+                            new GeoSearchResultOptions(SortOrder.ASC, 2));
+        }
+
+        var expectedResults =
+                new Object[] {
+                    2L, // geoadd(geoKey1, Map.of("Palermo", ..., "Catania", ...))
+                    new Double[][] {
+                        {13.36138933897018433, 38.11555639549629859},
+                        {15.08726745843887329, 37.50266842333162032},
+                    }, // geopos(geoKey1, new String[]{"Palermo", "Catania"})
+                    166274.1516, // geodist(geoKey1, "Palermo", "Catania")
+                    166.2742, // geodist(geoKey1, "Palermo", "Catania", GeoUnit.KILOMETERS)
+                    new String[] {
+                        "sqc8b49rny0", "sqdtr74hyu0", null
+                    } // eohash(geoKey1, new String[] {"Palermo", "Catania", "NonExisting"})
+                };
+
+        if (REDIS_VERSION.isGreaterThanOrEqualTo("6.2.0")) {
+            expectedResults =
+                    concatenateArrays(
+                            expectedResults,
+                            new Object[] {
+                                new String[] {
+                                    "Palermo", "Catania"
+                                }, // geosearch(geoKey1, "Palermo", byradius(200, km))
+                                new String[] {
+                                    "Palermo", "Catania"
+                                }, // geosearch(geoKey1, (15,37), bybox(200,200,km))
+                                new Object[] {
+                                    new Object[] {
+                                        "Palermo",
+                                        new Object[] {
+                                            0.0, 3479099956230698L, new Object[] {13.361389338970184, 38.1155563954963}
+                                        }
+                                    },
+                                    new Object[] {
+                                        "Catania",
+                                        new Object[] {
+                                            166.2742,
+                                            3479447370796909L,
+                                            new Object[] {15.087267458438873, 37.50266842333162}
+                                        }
+                                    }
+                                },
```

Comment:
Add comments to results.
Optional: you can define variables with result objects.

Reply:
since these don't get reused, what would be the purpose of creating variables for the geosearch results?

Reply:
Just readability. Up to you.